splash/network_manager.py

```diff
@@ -226,7 +240,12 @@ def _handle_custom_headers(self, request):
             headers = headers.items()
 
         for name, value in headers or []:
-            request.setRawHeader(to_bytes(name), to_bytes(value))
+            try:
+                request.setRawHeader(to_bytes(name), to_bytes(value))
+            except TypeError:
+                msg = "invalid header {}: {}. Header keys and values must be string or bytes"
+                self.log(msg.format(name, value), min_level=1, format_msg=False)
+                continue
```

Comment:
My thinking here is that we'll have to do str() on those values. str() can throw some error again (e.g. someone can pass Lua function by accident or pass something else weird), so to be 100% sure no exception happens we'll have to wrap this call to str() in another try-except which may decrease code readability.

HTTP headers are strings, and users can easily fix errors on their side - just pass proper value, string not number.